### DISCUSSION TITLE
v19.29.1-4.0.1 - YTLitePlus

### DISCUSSION THREAD

**Discussion by Balackburn** [YTLitePlus/YTLitePlus#263] [Announcements]
# ⚠️ Must Read ⚠️
- The .ipa file is no longer shared on Github due to a [Google DMCA notice](https://www.reddit.com/r/jailbreak/comments/1b86tiz/balackburn_here_maintainer_of_ytliteplus_ive_just/) 
- You need to build it yourself. See [Wiki-Building](https://github.com/Balackburn/YTLitePlus/wiki/Building)

# YTLitePlus Release Information

**Current YouTube IPA:** `19.29.1`  
**Current YTLite Version:** `4.0.1`  
**Current App Compatibility:** `iOS/iPadOS 15.0` or later

### RELEASE 19.29.1-4.0.1 (2024-07-22)

- Updated YouTube from `19.23.3` to `19.29.1`

#### New Features
- Added `Hide Home Tab` option (@bhackel) [#210](#)
- Added `Fullscreen to the Right` option (@arichornlover)
- Added `Disable Engagement Overlay` option (@bhackel)
- Added `Classic Seek Anywhere` gesture (@bhackel)
- Added YTHeaders submodule (@Balackburn)

#### Bug Fixes
- Fixed escaping quotes issue (@AlexInABox) [#187](#)
- Fixed SDK caching issues (@arichornlover)
- Fixed crash caused by edge case (@bhackel)
- Fixed compilation errors (@bhackel)
- Fixed home contents bug (@bhackel)
- Fixed `Fullscreen to the Right` functionality (@bhackel)
- Fixed redefinition of -userInterfaceIdiom issue (@asdfzxcvbn)

#### Improvements
- Cleaned up and added workflow settings (@bhackel)
- Updated compiling for YouTube v19.21.2+ (@arichornloverALT)
- Removed deprecated and redundant code (@bhackel)
- Synchronized delete-workflow-runs.yml (@arichornlover)

#### Localization
- Updated German translation for "CHANGE_APP_ICON" string (@AlexInABox) [#186](#)
- Updated Portuguese localization (@VictorxD70) [#253](#), [#254](#)
- Updated Spanish localization (@Deci8BelioS) [#260](#)

#### Updated Submodules
- Updated submodules: protobuf, iSponsorBlock, Return-YouTube-Dislikes, YouPiP, YouQuality, YouTubeHeader, YTHoldForSpeed, YTUHD, YouGroupSettings, YouMute, DontEatMyContent (@actions-user)
- Replaced submodule at Tweaks/YouPiP with https://github.com/bhackel/YouPiP (@bhackel)

#### Miscellaneous
- Synchronized uYouEnhanced templates (@bhackel)

### Contributors
- @AlexInABox
- @Balackburn
- @arichornlover
- @bhackel
- @asdfzxcvbn
- @VictorxD70
- @Deci8BelioS

### Issues in this Release
- LowContrastMode does not work *(Compatible only with YouTube v17.33.2-v17.38.10)*
- Hiding Shorts creates an empty space instead (#55)

### Catbox
`3iwg0o.ipa`

<hr /><em>This discussion was created from the release <a href='https://github.com/YTLitePlus/YTLitePlus/releases/tag/v19.29.1-4.0.1'>v19.29.1-4.0.1 - YTLitePlus</a>.</em>

**Comment by aricloverEXTRA** [comment]
Ooo, this Release layout looks better. Good job Balackburn!

**Reply by Balackburn** [reply]
> Ooo, this Release layout looks better. Good job Balackburn!

I used this script to get all the commits since latest release and asked ChatGPT to write the changelog : 

```
import requests
import os
from typing import List, Dict, Set, Tuple

# GitHub API base URL
GITHUB_API_BASE_URL = "https://api.github.com"

def get_github_token() -> str:
    """Retrieve GitHub token from environment variable."""
    token = os.getenv("GITHUB_TOKEN")
    if not token:
        raise ValueError("GitHub token not found. Set the GITHUB_TOKEN environment variable.")
    return token

def get_latest_tag(owner: str, repo: str, headers: Dict[str, str]) -> str:
    """Get the latest tag from the repository."""
    url = f"{GITHUB_API_BASE_URL}/repos/{owner}/{repo}/tags"
    response = requests.get(url, headers=headers)
    response.raise_for_status()
    tags = response.json()
    if not tags:
        raise ValueError("No tags found in the repository.")
    return tags[0]['name']

def get_default_branch(owner: str, repo: str, headers: Dict[str, str]) -> str:
    """Get the default branch of the repository."""
    url = f"{GITHUB_API_BASE_URL}/repos/{owner}/{repo}"
    response = requests.get(url, headers=headers)
    response.raise_for_status()
    return response.json()['default_branch']

def get_commits(owner: str, repo: str, base: str, head: str, headers: Dict[str, str]) -> List[Dict]:
    """Get all commits between two references."""
    url = f"{GITHUB_API_BASE_URL}/repos/{owner}/{repo}/compare/{base}...{head}"
    response = requests.get(url, headers=headers)
    response.raise_for_status()
    return response.json()['commits']

def is_submodule_update(commit_message: str) -> bool:
    """Check if the commit is a submodule update."""
    return commit_message.strip().lower() == "updated submodules"

def get_submodule_updates(owner: str, repo: str, commit_sha: str, headers: Dict[str, str]) -> Set[str]:
    """Extract submodule updates from a commit."""
    url = f"{GITHUB_API_BASE_URL}/repos/{owner}/{repo}/commits/{commit_sha}"
    response = requests.get(url, headers=headers)
    response.raise_for_status()
    commit_data = response.json()
    
    updates = set()
    for file in commit_data['files']:
        if file['filename'].startswith('Tweaks/'):
            submodule = file['filename'].split('/')[1]
            updates.add(submodule)
    return updates

def generate_release_notes(owner: str, repo: str, default_branch: str = None) -> str:
    """Generate release notes for the repository."""
    token = get_github_token()
    headers = {
        "Authorization": f"token {token}",
        "Accept": "application/vnd.github.v3+json"
    }

    if not default_branch:
        default_branch = get_default_branch(owner, repo, headers)

    latest_tag = get_latest_tag(owner, repo, headers)
    commits = get_commits(owner, repo, latest_tag, default_branch, headers)

    release_notes = ["Release Notes:"]
    submodule_updates: List[Tuple[str, Set[str]]] = []
    
    for commit in commits:
        message = commit['commit']['message'].split('\n')[0]  # Get first line of commit message
        author = commit['author']['login'] if commit['author'] else "Unknown"
        if is_submodule_update(message):
            updates = get_submodule_updates(owner, repo, commit['sha'], headers)
            submodule_updates.append((author, updates))
        elif not message.lower().startswith('updated'):  # Exclude individual "Updated X submodule" commits
            release_notes.append(f"- {message} (@{author})")

    if submodule_updates:
        release_notes.append("- Updated submodules:")
        all_updates = set()
        for author, updates in submodule_updates:
            all_updates.update(updates)
            release_notes.append(f"  - By @{author}:")
            for submodule in sorted(updates):
                release_notes.append(f"    - {submodule}")

    return "\n".join(release_notes)

def main():
    owner = input("Enter the repository owner: ")
    repo = input("Enter the repository name: ")
    default_branch = input("Enter the default branch name (press Enter for auto-detect): ").strip() or None

    try:
        release_notes = generate_release_notes(owner, repo, default_branch)
        print(release_notes)
    except requests.RequestException as e:
        print(f"Error communicating with GitHub API: {e}")
    except ValueError as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()
```

**Comment by backstab5983** [comment]
Thanks to everyone that worked on this!

**Reply by JudeTheTechSavvyDude** [reply]
It's still showing no icons ![image](https://github.com/user-attachments/assets/9245c98a-c866-4615-ad8c-062de6b215f6)

**Reply by aricloverEXTRA** [reply]
I thought it was patched? Did the No Icons bug came back?

**Reply by Balackburn** [reply]
> I thought it was patched? Did the No Icons bug came back?

I don't have it myself can't reproduce

**Reply by backstab5983** [reply]
> I thought it was patched? Did the No Icons bug came back?

Looks patched to me, can't reproduce.

**Reply by bhackel** [reply]
@JudeTheTechSavvyDude Can you export your AB settings and copy them into a pastebin/gist, maybe there is a flag for the issue

**Reply by JudeTheTechSavvyDude** [reply]
Should i uninstall it, then reinstall it, before i do the A/B

**Reply by bhackel** [reply]
I think not, since that may fix the issue, and I want flags that have the problem. Most of the settings should not interfere so no need to worry about that.

**Reply by JudeTheTechSavvyDude** [reply]
I think i fixed it already, it was a icon issue

**Comment by binnichtaktiv** [comment]
Thank you 🫶🏻 🫶🏻

**Comment by abheizthe** [comment]
How to "fullscreen to the right" - always emter fullscreen with home button on the right side?

**Reply by aricloverEXTRA** [reply]
> How to "fullscreen to the right" - always emter fullscreen with home button on the right side?

It should automatically set the fullscreen to the right once you enter fullscreen while the option is enabled in settings. Is it not working?

**Reply by abheizthe** [reply]
> 
> It should automatically set the fullscreen to the right once you enter fullscreen while the option is enabled in settings. Is it not working?

Yes, it's not working. Did removing ytabconfig causing this?

**Comment by aricloverEXTRA** [comment]
Copy & Paste buttons were adding in qnblackcat's uYouPlus now which can be useful!
https://github.com/qnblackcat/uYouPlus/commit/14a7123b39dc0412aee3e334a9df34120da947eb
It's great that it's not only an uYouEnhanced/YTLitePlus-only feature.

**Comment by CSummerChild** [comment]
I didn't realize this until now, and I'll make an issue ticket soon. But when clicking on community images it won't show the full thing.

**Reply by bhackel** [reply]
Yes #265

**Comment by truemanfree** [comment]
I love this release. Thank you.

**Reply by yumii550** [reply]
this is the best in my opinion because the last one u cant copy and past text and there have to many bugs
